Shopify/shopify-cli
# Pre-populate GitHub Issues when User Reports an Error

### WHY are these changes introduced?

Fixes #55 https://github.com/Shopify/shopify-cli-planning/issues/55
Part of https://github.com/Shopify/shopify-cli-planning/issues/22

When users encounter an error, they can directly open a GitHub issue template with pre-populated stacktrace information.

### WHAT is this pull request doing?
Cmd + press `submit an issue` to open GitHub Issues:
<img width="643" alt="Screen Shot 2021-10-19 at 1 48 49 PM" src="https://user-images.githubusercontent.com/55213953/137964130-43990e30-9f99-41d5-94bc-1a020063dc9b.png">

Pre-populates up to first 5 lines of stacktrace:
<img width="726" alt="Screen Shot 2021-10-19 at 1 49 10 PM" src="https://user-images.githubusercontent.com/55213953/137964059-0bf0b175-1402-4a33-9081-f08d145f494d.png">

### How to test your changes?
1. Modify the code to always raise an error by editing `lib/shopify_cli/commands/whoami.rb`. Add this on **line 7**:
    ```
    raise "Fail!"
    ```
2. Using iTerm or MacOS Terminal, execute command `SHOPIFY_CLI_DEVELOPMENT=0 bin/shopify whoami`, from the project root. This should display an error like the screenshot above. (Some terminals don't support links.)
3. Cmd + click on the **submit an issue** link and ensure that a new browser window is opened with a new Github issue
4. Check the contents of the pre-populated body, ensure that the last 5 stack trace is included

### Post-release steps



### Update checklist
